src/lib.rs

```diff
@@ -103,11 +103,15 @@ pub enum Error {
     /// });
     /// ```
     Syntax(Box<regex_syntax::Error>),
+
+    /// Example of an unsatisfiable regex: `[a&&b]`
```

Comment:
``````suggestion
    /// The regex can never be matched, and thus it is impossible to generate 
    /// any samples from the regex.
    ///
    /// # Examples
    ///
    /// ```
    /// let err = rand_regex::Regex::compile(r"[a&&b]", 100).unwrap_err();
    /// assert_eq!(gen, rand_regex::Error::Unsatisfiable);
    /// ```
``````